src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,192 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> node(int index) {
+        checkElementIndex(index);
+        if (index < (size / 2)) {
+            Node<T> x = first;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = last;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+            return x;
+        }
+    }
+
+    private void addFirst(T value) {
+        Node<T> f = first;
+        Node<T> newNode = new Node<>(f, value, null);
+        first = newNode;
+        if (f == null) {
+            last = newNode;
+        } else {
+            f.prev = newNode;
+        }
+        size++;
+    }
+
+    private void addLast(T value) {
+        Node<T> l = last;
+        Node<T> newNode = new Node<>(l, value, null);
```

Comment:
Consider explicitly setting the `next` reference of the new node to `null` for clarity, even though it defaults to `null`.